### PR TITLE
C-007 kukai deprecation

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/Development-05abe2b65cd14ae39a101533bc01b708?p=ca50c07d2e4744ac95509c8d9b8b4530

### CODE REVIEW

File: src/utils/kukai.utils.ts [hunk 1]
```diff
@@ -17,6 +17,13 @@ const decrypt = async (chiphertext: string, password: string, salt: string, vers
   }
 };
 
+export class KukaiError extends Error {
+  constructor() {
+    super();
+    throw new Error(KUKAI_VERSION_ERROR);
```
Comment (unixvb) [inline src/utils/kukai.utils.ts:23]: I'm not sure that throwing a new Error inside `constructor` is the right way to create a custom error.  
mb just set a correct message?
image

Comment (letier3110) [reply src/utils/kukai.utils.ts:23]: but its typescript, not a esjs

Comment (unixvb) [reply src/utils/kukai.utils.ts:23]: so, you could google solution for ts
https://ashsmith.io/handling-custom-error-classes-in-typescript
```
class MyError extends Error {
  constructor() {
    super('your message');
    Object.setPrototypeOf(this, MyError.prototype)
  }
}
```

**Review by unixvb — APPROVED** [review]

File: src/utils/kukai.utils.ts [hunk 2]
```diff
@@ -55,12 +44,15 @@ const bumpIV = (salt: string, bumps: number) => {
 
 export const decryptSeedPhrase = async (json: string, pwd: string) => {
   const walletData = JSON.parse(json);
+  if (walletData.walletType === 4 && (walletData.version === 1 || walletData.version === 2)) {
```
Comment (unixvb) [inline src/utils/kukai.utils.ts:47]: are you sure that only `walletType: 4` should not be decrypted? mb remove that condition and check only `various`

**Pull request merged**